src/Features/Core/Portable/Diagnostics/Analyzers/IDEDiagnosticIds.cs

```diff
@@ -66,6 +66,13 @@ internal static class IDEDiagnosticIds
 
         public const string MakeFieldReadonlyDiagnosticId = "IDE0044";
 
+        public const string RegexPatternDiagnosticId = "IDE0045";
+
+        // diagnostic for when we detect a string that should have /*language=json*/ added to it.
+        public const string JsonDetectionDiagnosticId = "IDE0046";
+        // diagnostic for when we have a known json string and we detect something wrong with it.
+        public const string JsonPatternDiagnosticId = "IDE0047";
```

Comment:
I wasn't able to get squiggles for bad Json in the IDE, with this error code. Do I need to turn some option on?

Reply:
> pdate: I was able to see squiggles in string s = /*lang=json*/ "{a: 1, b: new()}"; under the open paren. But it's barely visible.

What's the experience in the IDE elsewhere for this sort of error?  I'm not sure how to necessarily make that better if we're *only* squiggling the open paren.  That said, we could squiggle more, and thus make the squiggle more visible.

> I was able to see the QuickInfo with the diagnostic, but it only appears if your cursor doesn't accidentally trigger any other QuickInfo. Is that by design?

Yeah, you can get the same QI behavior today in other areas of your code.  However, things are very exacerbated with regex/json strings since you're always hovering over part of a string.  I think i should probably suppress normal QI for these strings so you don't have the issue you've identified.  Great catch!